mtags/src/main/scala-2/scala/meta/internal/pc/InferredTypeProvider.scala

```diff
@@ -28,10 +29,14 @@ final class InferredTypeProvider(
 ) {
   import compiler._
 
-  def inferredTypeEdits(): List[TextEdit] = {
+  def inferredTypeEdits(
+      text: Option[String] = None,
+      adjustedEndPos: Option[l.Position] = None,
+      retryType: Boolean = true
```

Comment:
Can we group these parameters in a case class like

```scala
case class AdjustTypeOpts(
  text: String,
  adjustedEndPos: l.Position
)

def inferredTypeEdits(
  adjustOpt: Option[AdjustTypeOpts] = None
) = {
  val retryType = adjustOpt.isEmpty
}
```

?

Those parameters are always

- `None, None, true` for the first try, or
- `Some(...), Some(...), false` when it's retry

if we group them up, we can easily know the parameters like `Some(...), None, true` won't be happen.

Reply:
BTW, removing the `: Type` part and retrying to insert inferred type is really good idea! Great job :+1:

Reply:
> BTW, removing the `: Type` part and retrying to insert inferred type is really good idea! Great job +1

I actually tried getting it from RHS, but the type was just error.

Reply:
I applied the refactoring, thanks for the suggestion!